File: mifosng-android/src/main/java/com/mifos/mifosxdroid/formwidgets/FormToggleButton.java
package com.mifos.mifosxdroid.formwidgets;

import android.content.Context;
import androidx.appcompat.widget.SwitchCompat;
import android.view.Gravity;
import android.widget.CompoundButton;
import android.widget.LinearLayout;
import android.widget.TextView;

/**
 * Created by Tarun on 1/28/2017.
 */
public class FormToggleButton extends FormWidget {
    protected TextView label;
    protected SwitchCompat switchButton;

    private LinearLayout.LayoutParams weightedLayoutParams =
            new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,
                    LinearLayout.LayoutParams.WRAP_CONTENT, (float) 0.5);

    private boolean isTrue = false;

    public FormToggleButton(Context context, String name) {

        super(context, name);

        LinearLayout linearLayout = new LinearLayout(context);
        linearLayout.setLayoutParams(FormWidget.defaultLayoutParams);
        linearLayout.setOrientation(LinearLayout.HORIZONTAL);
        linearLayout.setGravity(Gravity.CENTER_HORIZONTAL);

        label = new TextView(context);
        label.setText(getDisplayText());
        label.setLayoutParams(weightedLayoutParams);
        linearLayout.addView(label);

        switchButton = new SwitchCompat(context);
        switchButton.setLayoutParams(weightedLayoutParams);
        switchButton.setGravity(Gravity.CENTER_HORIZONTAL);
        switchButton.setSwitchMinWidth(50);
        switchButton.setShowText(true);
        switchButton.setChecked(false);
        switchButton.setTextOn("True");
        switchButton.setTextOff("False");
        switchButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                isTrue = b;
            }
        });
        linearLayout.addView(switchButton);

        layout.addView(linearLayout);

    }

    @Override
    public String getValue() {
        return isTrue ? "true" : "false";
    }

}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/formwidgets/FormToggleButton.kt
package com.mifos.mifosxdroid.formwidgets

import android.content.Context
import android.view.Gravity
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.widget.SwitchCompat

/**
 * Created by Tarun on 1/28/2017.
 */
class FormToggleButton(context: Context?, name: String?) : FormWidget(context, name) {
    private var label: TextView
    private var switchButton: SwitchCompat
    private val weightedLayoutParams = LinearLayout.LayoutParams(
        LinearLayout.LayoutParams.MATCH_PARENT,
        LinearLayout.LayoutParams.WRAP_CONTENT, 0.5.toFloat()
    )
    private var isTrue = false

    init {
        val linearLayout = LinearLayout(context)
        linearLayout.layoutParams = defaultLayoutParams
        linearLayout.orientation = LinearLayout.HORIZONTAL
        linearLayout.gravity = Gravity.CENTER_HORIZONTAL
        label = TextView(context)
        label.text = displayText
        label.layoutParams = weightedLayoutParams
        linearLayout.addView(label)
        switchButton = SwitchCompat(context!!)
        switchButton.layoutParams = weightedLayoutParams
        switchButton.gravity = Gravity.CENTER_HORIZONTAL
        switchButton.switchMinWidth = 50
        switchButton.isChecked = false
        switchButton.setOnCheckedChangeListener { compoundButton, b -> isTrue = b }
        linearLayout.addView(switchButton)
        layout.addView(linearLayout)
    }

    override var value: String
        get() = if (isTrue) "true" else "false"
        set(value) {}
}